src/Microsoft.PowerShell.Commands.Utility/commands/utility/ShowMarkdownCommand.cs

```diff
@@ -61,6 +62,8 @@ public string[] LiteralPath
 
         private System.Management.Automation.PowerShell _powerShell;
 
+        private readonly List<string> _inputObjectBuffer = new();
```

Comment:
Was considering using `StringBuilder` here instead but kept it as `List` since `ConvertFrom-Json` uses it for buffer.

Reply:
I also think for this use case string builder is not more efficient than list + `string.Join`, but maybe can do benchmark to be sure.

Reply:
@iSazonov thoughts?

Reply:
I guess Markdown formatting by line so we have no need to collect input to one string, yes?

Reply:
In this case it is required to collect input in one string so `Show-Markdown` can work with array of markdown lines being passed in to `-InputObject`. This is similar to how `ConvertFrom-Json` works with input buffer, it collects all input and processes it once at the end.

Reply:
Json is multi-string. Markdown process text string by string, yes?

Reply:
I'm not sure if I'm following. Markdown cmdlet originally could only process single string. So if you had an array of strings, e.g. what `Get-WhatsNew` markdown returns, `Show-Markdown` couldn't process it, since it needs full markdown string, and not one line at a time. This change just processes all lines at once from buffer, making it work with single string or array of strings.

Had a discussion with @mklement0 in another issue: https://github.com/PowerShell/whatsnew/issues/23#issuecomment-1867036757 about proposed fix.

Reply:
Since base cmdlet doesn't support "streaming" conversion I close my question. 
Without "streaming" conversion we have performance issue with large texts and scenarios like `Get-Content | Show-Markdown`. The best thing would be if `MarkdownConverter.Convert()` accepted a parameter of type Collection/List.

Reply:
I see MarkdownConverter.Convert() accept only string. So current code is too optimistic to send `object` to ConvertFrom-Markdown. It is better to convert all to string here.